afs-core/src/main/java/com/powsybl/afs/AbstractNodeBase.java

```diff
@@ -175,16 +185,47 @@ private boolean nodeNameAlreadyExists(String name) {
         return childNodes.stream().filter(nodeInfo -> !nodeInfo.getId().equals(getId())).anyMatch(nodeInfo -> nodeInfo.getName().equals(name));
     }
 
-    public void archive(Path dir) {
+    public void archive(Path dir, boolean useZip){
+
+        //if it's a copy paste, this test is already done
+        if(useZip) {
+            File archiveFile = dir.toFile();
+            long freeSpacePercent = 100 * archiveFile.getFreeSpace() / archiveFile.getTotalSpace();
+            LOGGER.info("Copying into drive with {}% free space", freeSpacePercent);
+            if (freeSpacePercent < MIN_DISK_SPACE_THRESHOLD) {
+                throw new AfsException("Not enough space");
+            }
+        }
+
         Objects.requireNonNull(dir);
         try {
             new AppStorageArchive(storage).archive(info, dir);
+            if(useZip){
+                Path zipPath = dir.resolve(info.getId() + ".zip");
+                Utils.zip(dir.resolve(info.getId()), zipPath);
+            }
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
-    public void unarchive(Path dir) {
-        new AppStorageArchive(storage).unarchive(info, dir);
+    public void unarchive(Path dir, boolean isZipped) {
+        if(isZipped){
+            int index = dir.toString().lastIndexOf(".");
+            Path nodeDir = Paths.get(dir.toString().substring(0, index));
+            Utils.unzip(dir, nodeDir);
+            new AppStorageArchive(storage).unarchive(info, nodeDir);
+            Utils.deleteDirectory(new File(nodeDir.toString()));
```

Comment:
make a `try..catch..finally` with uncompressed directory deletion on the finally clause

Reply:
well, just a `try..finally` since we would want to rethrow the exception